src/Interface/Modules/Render/ViewScene.cc

```diff
@@ -149,8 +149,166 @@ namespace Gui {
+{
+  return previousCameraTransform_ == mat;
+}
+
+//--------------------------------------------------------------------------------------------------
+void PreviousWidgetSelectionInfo::widgetColorRestored()
+{
+  timeWidgetColorRestored_ = std::chrono::system_clock::now();
+}
+
+//--------------------------------------------------------------------------------------------------
+void PreviousWidgetSelectionInfo::selectionAttempt()
+{
+  timeOfLastSelectionAttempt_ = std::chrono::system_clock::now();
+}
+
+//--------------------------------------------------------------------------------------------------
+void PreviousWidgetSelectionInfo::setCameraTransform(glm::mat4 mat)
+{
+  previousCameraTransform_ = mat;
+}
+
+//--------------------------------------------------------------------------------------------------
+void PreviousWidgetSelectionInfo::setMousePosition(int x, int y)
+{
+  lastMousePressEventX_ = x;
+  lastMousePressEventY_ = y;
+}
+
+//--------------------------------------------------------------------------------------------------
+void PreviousWidgetSelectionInfo::setFrameIsFinished(bool finished)
+{
+  frameIsFinished_ = finished;
+}
+
+//--------------------------------------------------------------------------------------------------
+bool PreviousWidgetSelectionInfo::getFrameIsFinished() const
+{
+  return frameIsFinished_;
+}
+
+//--------------------------------------------------------------------------------------------------
+long PreviousWidgetSelectionInfo::timeSinceWidgetColorRestored() const
+{
+  return timeSince(timeWidgetColorRestored_);
+}
+
+//--------------------------------------------------------------------------------------------------
+long PreviousWidgetSelectionInfo::timeSinceLastSelectionAttempt() const
+{
+  return timeSince(timeOfLastSelectionAttempt_);
+}
+
+//--------------------------------------------------------------------------------------------------
+void PreviousWidgetSelectionInfo::setPreviousWidget(const WidgetHandle widget)
+{
+  previousSelectedWidget_ = widget;
+}
+
+//--------------------------------------------------------------------------------------------------
+WidgetHandle PreviousWidgetSelectionInfo::getPreviousWidget() const
+{
+  return previousSelectedWidget_;
+}
+
+//--------------------------------------------------------------------------------------------------
+bool PreviousWidgetSelectionInfo::hasSameWidget(const WidgetHandle widget) const
+{
+  return previousSelectedWidget_ == widget;
+}
+
+//--------------------------------------------------------------------------------------------------
+bool PreviousWidgetSelectionInfo::hasPreviousWidget() const
+{
+  if (previousSelectedWidget_)
+    return true;
+  else
+    return false;
+}
+
+//--------------------------------------------------------------------------------------------------
+bool PreviousWidgetSelectionInfo::deletePreviousWidget()
+{
+  previousSelectedWidget_.reset();
```

Comment:
@RubioJr9 `error C4716: 'SCIRun::Gui::PreviousWidgetSelectionInfo::deletePreviousWidget': must return a value `

Reply:
Sorry, I'm not sure how that didn't trip up my compiler on mac os